test : improve mempool_accept_wtxid.py

naiza2000 opened the pull request:
Follow-up to #22253 adding changes suggested in [#22253 (review)](https://github.com/bitcoin/bitcoin/pull/22253#discussion_r666933370)

glozow commented:
Concept ACK, thanks for picking this up! Needs a [rebase and squash](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits).

sriramdvt commented on test/functional/mempool_accept_wtxid.py:
```diff
@@ -79,10 +79,7 @@ def run_test(self):
-        child_two.vin.append(CTxIn(COutPoint(int(parent_txid, 16), 0), b""))
-        child_two.vout.append(CTxOut(int(9.99996 * COIN), child_script_pubkey))
-        child_two.wit.vtxinwit.append(CTxInWitness())
+        child_two = copy.deepcopy(child_one)
```
Importing `copy` using `import copy` at the start might solve the failing tests. It might be a better idea to import only deepcopy using `from copy import deepcopy` and then using the function directly here.
```suggestion
        child_two = deepcopy(child_one)
```
There are some other errors that I am getting in "Submit child_two to the mempool" even after making this change on my local machine.

brunoerg commented:
@michaelfolkson gave me a tip in my first PR that might be helpful for you:

You can do a reset on your branch to the current top commit on master, then re-adding your commit and force pushing.

git reset --hard insert_current_top_commit_hash_on_master

Then add a single commit on top and force pushing

naiza2000 committed:
Improve mempool_accept_wtxid.py

Improve mempool_accept_wtxid.py

Improve mempool_accept_wtxid.py

Improve mempool_accept_wtxid.py

Improve mempool_accept_wtxid.py

jnewbery commented:
Concept ACK

Did you consider also doing:

> I think the simplest parent witness script to create same-nonwitness-data txs would be OP_DROP OP_TRUE, with a different arbitrary witness stack element in each child transaction. OTOH multiple script branches are far more realistic, I guess that's the primary scenario :)

(https://github.com/bitcoin/bitcoin/pull/22253#pullrequestreview-703027841)

theStack commented:
Concept ACK

Thanks for picking this up and welcome as a new contributor! 🎉 
Happy to review/test as soon as the CI is green :)

glozow commented on test/functional/mempool_accept_wtxid.py:
```diff
@@ -4,9 +4,9 @@
+with identical non-witness data but different witness.
 """
-
+import copy
```
Per [PEP8](https://www.python.org/dev/peps/pep-0008/#imports), you want a line between standard library and local imports.

Also, as sriramdvt mentioned earlier, a more granular import of just `deepcopy` would be preferred.

glozow commented on test/functional/mempool_accept_wtxid.py:
```diff
@@ -104,13 +101,12 @@ def run_test(self):
         }])
-        testres_child_two = node.testmempoolaccept([child_two.serialize().hex()])[0]
-        assert_equal(testres_child_two, {
+        assert_equal(node.testmempoolaccept([child_two.serialize().hex()]), [{
```
Keeping the index would reduce the diff
```suggestion
        assert_equal(node.testmempoolaccept([child_two.serialize().hex()])[0], {
```

naiza2000 replied:
Done!

glozow commented:
(Also needs a squash)

glozow reviewed (commented):
utACK 91b05974fc1ea38062f12d36152201af81bda1a2

theStack reviewed (commented):
post-merge ACK 91b05974fc1ea38062f12d36152201af81bda1a2 🍺 

nit: not a big issue, but next time, please take care of the commit message body, the 4x"Improve mempool_accept_wtxid.py" doesn't contain any information :)

naiza2000 commented:
> post-merge ACK [91b0597](https://github.com/bitcoin/bitcoin/commit/91b05974fc1ea38062f12d36152201af81bda1a2) 🍺
> 
> nit: not a big issue, but next time, please take care of the commit message body, the 4x"Improve mempool_accept_wtxid.py" doesn't contain any information :)

Yes, thank you for the guidance. Will certainly take care of that in the future.
